### DISCUSSION TITLE
How does xournalpp register pen events?

### DISCUSSION THREAD

**Discussion by SunkenHero** [xournalpp/xournalpp#4325] [Q&A]
So i want to know how xournalpp registers pen events. I already looked throught the code trying to find the code for it, but i couldnt find it. Mayby someone know where it is and can help me.

**Comment by rolandlo** [comment]
All input events are registered in `InputContext::handle` (file `src/core/gui/inputdevices/InputContext.cpp`).

**Comment by SunkenHero** [comment]
Thank you for your help, this is what i searched for.
